### PR TITLE
feat(client): add app linking | db linking functionality

### PR DESCRIPTION
**Pull request by Akirtovskis**
@pradel basic functionality for linking works. 

 Couple of issues that I will open once this is approved : 
 - feedback after successful link ( some sort of green snackbar seems good to me ) 
 - handling db deletion if db is still linked  ( for app deletion it works even if app is linked it just shows "unlinking and then proceeds w deletion")
 - also list of connected apps / dbs so you don't see the connected apps in dropdown, but rather outside
 - making dropdown menu a separate component perhaps? ( but this seems kind of overkill at least for now ) 

Couple of screenshots : 

Screenshot 2020-08-03 at 20 06 13
Screenshot 2020-08-04 at 11 52 25
Screenshot 2020-08-03 at 20 12 44

### CODE REVIEW

File: server/src/graphql/mutations/linkDatabase.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    throw new Error(
+      `Database with ID ${databaseId} does not belong to ${userId}`
+    );
+  }
+
+  const dbType = dbTypeToDokkuPlugin(database.type);
+
+  const ssh = await sshConnect();
+
+  const isLinked = await dokku.database.linked(
```
Comment (pradel) [inline server/src/graphql/mutations/linkDatabase.ts:54]: you can also check if the link exist with a db query, it might actually be faster :)

Comment (Akirtovskis) [reply server/src/graphql/mutations/linkDatabase.ts:54]: @pradel nice will get rid of this, if we can just check that ad not via dokku, then it is amazing as it makes couple of things easier in my next PR.

File: server/src/graphql/queries/isDatabaseLinked.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    where: {
+      id: appId,
+    },
+  });
+
+  if (!app) {
+    throw new Error(`App with ID ${appId} not found`);
+  }
+
+  if (app.userId !== userId) {
```
[the remark this answers is not in the document]
Comment (Akirtovskis) [reply server/src/graphql/queries/isDatabaseLinked.ts:32]: @pradel totally makes sense.

**Comment by Akirtovskis** [comment]
@pradel feedbacks solved. Let me know if I missed something or something could still be improved. :)

File: client/src/pages/database/index.tsx [hunk 3]
```diff
@@ -27,12 +47,28 @@ export const Database = () => {
...
+      await linkDatabaseMutation({
+        variables: {
+          input: {
+            databaseId,
+            appId,
+          },
+        },
+      });
+    } catch (e) {
+      //
```
[the remark this answers is not in the document]
Comment (pradel) [reply client/src/pages/database/index.tsx:68]: maybe we can use https://github.com/fkhadra/react-toastify for showing errors? Unless you know a better one :)

**Comment by Akirtovskis** [comment]
@pradel new dropdown menus are in town. 

Screenshot 2020-08-06 at 14 16 42

**Comment by pradel** [comment]
@Akirtovskis let's pin the docs deps? And update them?

File: website/package.json [hunk 4]
```diff
@@ -9,12 +9,12 @@
...
   },
   "dependencies": {
-    "@docusaurus/core": "^2.0.0-alpha.54",
-    "@docusaurus/preset-classic": "^2.0.0-alpha.54",
-    "classnames": "^2.2.6",
-    "react": "^16.8.4",
-    "react-dom": "^16.8.4",
-    "react-responsive-carousel": "^3.2.9"
+    "@docusaurus/core": "2.0.0-alpha.60",
+    "@docusaurus/preset-classic": "v2.0.0-alpha.60",
```
Comment (pradel) [inline website/package.json:13]: should the "v" be removed? 🤔

File: server/src/graphql/queries/isDatabaseLinked.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        apps: {
+          where: {
+            id: appId,
+          },
+        },
+      },
+    })
+    .apps();
+
+  console.log(linkedApps);
```
Comment (pradel) [inline server/src/graphql/queries/isDatabaseLinked.ts:28]: to remove 😁

File: server/src/graphql/queries/isDatabaseLinked.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+export const isDatabaseLinked: QueryResolvers['isDatabaseLinked'] = async (
+  _,
+  { databaseId, appId },
+  { userId }
+) => {
+  if (!userId) {
+    throw new Error('Unauthorized');
+  }
+  // We query for link between particular db and particular app
+  const linkedApps = await prisma.database
```
Comment (pradel) [inline server/src/graphql/queries/isDatabaseLinked.ts:13]: did you test this logic to be sure it's working properly?

File: client/src/pages/database/index.tsx [hunk 7]
```diff
@@ -27,12 +45,35 @@ export const Database = () => {
...
+  const appOptions = apps.map((app) => {
+    return {
+      value: { name: app.name, id: app.id },
+      label: app.name,
+    };
+  });
+
+  const handleConnect = async (databaseId: string, appId: string) => {
+    try {
+      await linkDatabaseMutation({
```
Comment (pradel) [inline client/src/pages/database/index.tsx:64]: do we want to show some kind of feedback that this worked at the end? can be a TODO TOASTIFY :p

Comment (Akirtovskis) [reply client/src/pages/database/index.tsx:64]: @pradel Yes, that's a great idea. TODO added.

File: server/src/graphql/queries/isDatabaseLinked.ts (hunk 6 above)
Comment (Akirtovskis) [reply server/src/graphql/queries/isDatabaseLinked.ts:13]: @pradel tested this and is working!

**Pull request merged**